utils/service_discovery/sd_docker_backend.py

```diff
@@ -242,18 +242,33 @@ def get_tags(self, state, c_id):
                 log.warning("Failed to fetch pod metadata for container %s."
                             " Kubernetes tags may be missing." % c_id[:12])
                 return []
-            # get labels
+
+            # get pod labels
             kube_labels = pod_metadata.get('labels', {})
             for label, value in kube_labels.iteritems():
                 tags.append('%s:%s' % (label, value))
 
-            # get replication controller
-            created_by = json.loads(pod_metadata.get('annotations', {}).get('kubernetes.io/created-by', '{}'))
-            if created_by.get('reference', {}).get('kind') == 'ReplicationController':
-                tags.append('kube_replication_controller:%s' % created_by.get('reference', {}).get('name'))
-
             # get kubernetes namespace
-            tags.append('kube_namespace:%s' % pod_metadata.get('namespace'))
+            namespace = pod_metadata.get('namespace')
+            tags.append('kube_namespace:%s' % namespace)
+
+            # get created-by
+            created_by = json.loads(pod_metadata.get('annotations', {}).get('kubernetes.io/created-by', '{}'))
+            creator_kind = created_by.get('reference', {}).get('kind')
+            creator_name = created_by.get('reference', {}).get('name')
+
+            # add creator tags
+            if creator_kind == 'ReplicationController':
+                tags.append('kube_replication_controller:%s' % creator_name)
```

Comment:
`creator_name` may be `None`, we should check it if we don't want to create tag like `kube_replication_controller:None`